tools/clang/lib/CodeGen/CGHLSLMSFinishCodeGen.cpp

```diff
@@ -2824,7 +2834,25 @@ void TranslateRayQueryConstructor(HLModule &HLM) {
         llvm::ConstantInt::get(i32Ty, (uint64_t)0, false);
     llvm::FunctionType *funcTy =
         llvm::FunctionType::get(i32Ty, {i32Ty, i32Ty}, false);
+
     unsigned opcode = (unsigned)IntrinsicOp::IOP_AllocateRayQuery;
+    // reset opcode to IOP_AllocateRayQuery2 if there are 2 template args
```

Comment:
There is no IOP_AllocateRayQuery2 now.  Also, I believe this would be simpler if we always generated a single signature for the HL op, supplying both flag arguments, whether or not RayQueryFlags are provided in the template arguments or equal to zero.

Then, HLOperationLower can choose the DXIL op based on whether RayQueryFlags are zero at that point.